src/hooks/useElementSize.ts

```diff
@@ -0,0 +1,31 @@
+import { RefObject, useCallback, useEffect, useMemo, useRef, useState } from 'react';
+
+interface Rect {
+    width: number;
+    height: number;
+}
+
+export const useElementSize = <T extends HTMLDivElement>(): [RefObject<T>, Rect] => {
+    const ref = useRef<T>(null);
+    const [rect, setRect] = useState<Rect>({ width: 0, height: 0 });
+
+    const handleResize = useCallback(() => {
+        if (ref.current) {
+            setRect({ width: ref.current.offsetWidth, height: ref.current.offsetHeight });
+        }
+    }, []);
```

Comment:
we can create this function inside useMemo

Reply:
done